orttraining/orttraining/python/training/onnxblock/_building_blocks.py

```diff
@@ -0,0 +1,146 @@
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License.
+# _building_blocks.py
+
+import copy
+import onnx
+
+import onnxruntime.training.onnxblock as onnxblock
+import onnxruntime.training.onnxblock.model_accessor as accessor
+import onnxruntime.training.onnxblock._graph_utils as graph_utils
+
+
+class Sub(onnxblock.Model):
+    """Adds Sub node to an onnx model."""
+
+    def __init__(self):
+        super(Sub, self).__init__()
+
+    def build(self, sub_input_name1, sub_input_name2):
+        # get the model to manipulate
+        onnx_model = accessor.global_accessor.model
+
+        # create the graph node for sub
+        sub_node_input_names = [sub_input_name1, sub_input_name2]
+        sub_node_output_name = graph_utils.generate_random_graph_name("sub_output")
+        sub_node_output_names = [sub_node_output_name]
+        sub_node = onnx.helper.make_node(
+            "Sub",
+            sub_node_input_names,
+            sub_node_output_names,
+            name=graph_utils.generate_random_graph_name("Sub"),
+        )
+        onnx_model.graph.node.append(sub_node)
+
+        # create the graph output for sub
+        graph_output = copy.deepcopy(
+            graph_utils.get_output_from_output_name(onnx_model, sub_input_name1)
+        )
+        graph_output.name = sub_node_output_name
+        del onnx_model.graph.output[:]
+        onnx_model.graph.output.append(graph_output)
+
+        return sub_node_output_name
+
+
+class Pow(onnxblock.Model):
+    """Adds Pow node to the onnx model."""
+
+    def __init__(self, exponent):
+        super(Pow, self).__init__()
+
+        self._exponent = exponent
+
+    def build(self, pow_input_name):
+        # get the model to manipulate
+        onnx_model = accessor.global_accessor.model
+
+        # create the graph initializer for the exponent
+        pow_node_exponent_name = graph_utils.generate_random_graph_name("pow_exponent")
+        onnx_model.graph.initializer.append(
+            onnx.helper.make_tensor(
+                pow_node_exponent_name, onnx.TensorProto.FLOAT, [1], [self._exponent]
+            )
+        )
+
+        # create the graph node for pow
+        pow_node_input_names = [pow_input_name, pow_node_exponent_name]
+        pow_node_output_name = graph_utils.generate_random_graph_name("pow_output")
+        pow_node_output_names = [pow_node_output_name]
+        pow_node = onnx.helper.make_node(
+            "Pow",
+            pow_node_input_names,
+            pow_node_output_names,
+            name=graph_utils.generate_random_graph_name("Pow"),
+        )
+        onnx_model.graph.node.append(pow_node)
+
+        # create the graph output for pow
+        graph_output = copy.deepcopy(
+            graph_utils.get_output_from_output_name(onnx_model, pow_input_name)
+        )
+        graph_output.name = pow_node_output_name
```

Comment:
should we first check pow_input_name in the graph, then if it is one of the graph output, then we replace it with output of Pow. If it is not one of ghte graph output, then we need append a new graph output.

Reply:
In this scenario, if `pow_input_name` is not a graph output at this time, the graph building will fail.
Ideally, the graph output should be decided by the last layer of the graph building. Intermediate layers should not be concerned with making the graph outputs. I will address this in the next PR where I add the bce loss functionality.